simulation.hpp

```diff
@@ -129,6 +129,79 @@ class Simulation
         }
     }
 
+    /** Moves hosts from one location to another
+     *
+     * @param infected Currently infected hosts
+     * @param susceptible Currently susceptible hosts
+     * @param mortality_tracker Hosts that are infected at a specific time step
+     * @param total_plants 
+     * @param step the current step of the simulation
+     * @param last_index the last index to not be used from movements
+     * @param movements a vector of ints with row_from, col_from, row_to, col_to, and num_hosts
+     * @param movement_schedule a vector matching movements with the step at which the movement from movements are applied
+     */
+    unsigned movement(IntegerRaster& infected,
+                      IntegerRaster& susceptible, 
+                      IntegerRaster& mortality_tracker,
+                      IntegerRaster& total_plants,
+                      unsigned step, unsigned last_index,
+                      const std::vector<std::vector<int>>& movements,
+                      std::vector<unsigned> movement_schedule)
+    {
+        for (unsigned i = last_index; i < movements.size(); i++) {
+            auto moved = movements[i];
+            unsigned move_schedule = movement_schedule[i];
+            if (move_schedule != step) {
+                return i;
+            }
+            int infected_moved = 0;
+            int susceptible_moved = 0;
+            int total_hosts_moved = 0;
+            double infected_mean = 0;
+            double inf_ratio = 0;
+            int row_from = moved[0];
+            int col_from = moved[1];
+            int row_to = moved[2];
+            int col_to = moved[3];
+            int hosts = moved[4];
+            if (hosts > total_plants(row_from, col_from)) {
+                total_hosts_moved = total_plants(row_from, col_from);
+            } else {
+                total_hosts_moved = hosts;
+            }
+            if (infected(row_from, col_from) > 0 && susceptible(row_from, col_from) > 0) {
+                inf_ratio = double(infected(row_from, col_from)) / double(susceptible(row_from, col_from));
```

Comment:
This should actually be

    inf_ratio = double(infected(row_from, col_from)) / double(total_plants(row_from, col_from));

This will BTW get more complex with the exposed.